internal/activities/add_premis_event_test.go

```diff
@@ -0,0 +1,232 @@
+    <premis:eventDateTime></premis:eventDateTime>
+    <premis:eventDetailInformation>
+      <premis:eventDetail></premis:eventDetail>
+    </premis:eventDetailInformation>
+    <premis:eventOutcomeInformation>
+      <premis:eventOutcome>valid</premis:eventOutcome>
+    </premis:eventOutcomeInformation>
+    <premis:linkingAgentIdentifier>
+      <premis:linkingAgentIdentifierType valueURI="http://id.loc.gov/vocabulary/identifiers/local">url</premis:linkingAgentIdentifierType>
+      <premis:linkingAgentIdentifierValue>https://github.com/artefactual-sdps/preprocessing-sfa</premis:linkingAgentIdentifierValue>
+    </premis:linkingAgentIdentifier>
+  </premis:event>
+</premis:premis>
+`
+
+const expectedPREMISWithUnsuccessfulEvent = `<?xml version="1.0" encoding="UTF-8"?>
+<premis:premis xmlns:premis="http://www.loc.gov/premis/v3" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.loc.gov/premis/v3 https://www.loc.gov/standards/premis/premis.xsd" version="3.0">
+  <premis:object xsi:type="premis:file">
+    <premis:objectIdentifier>
+      <premis:objectIdentifierType>UUID</premis:objectIdentifierType>
+      <premis:objectIdentifierValue>52fdfc07-2182-454f-963f-5f0f9a621d72</premis:objectIdentifierValue>
+    </premis:objectIdentifier>
+    <premis:objectCharacteristics>
+      <premis:format>
+        <premis:formatDesignation>
+          <premis:formatName></premis:formatName>
+        </premis:formatDesignation>
+      </premis:format>
+    </premis:objectCharacteristics>
+    <premis:originalName>somefile.txt</premis:originalName>
+    <premis:linkingEventIdentifier>
+      <premis:linkingEventIdentifierType></premis:linkingEventIdentifierType>
+      <premis:linkingEventIdentifierValue></premis:linkingEventIdentifierValue>
+    </premis:linkingEventIdentifier>
+  </premis:object>
+  <premis:event>
+    <premis:eventIdentifier>
+      <premis:eventIdentifierType></premis:eventIdentifierType>
+      <premis:eventIdentifierValue></premis:eventIdentifierValue>
+    </premis:eventIdentifier>
+    <premis:eventType>someActivity</premis:eventType>
+    <premis:eventDateTime></premis:eventDateTime>
+    <premis:eventDetailInformation>
+      <premis:eventDetail></premis:eventDetail>
+    </premis:eventDetailInformation>
+    <premis:eventOutcomeInformation>
+      <premis:eventOutcome>invalid</premis:eventOutcome>
+    </premis:eventOutcomeInformation>
+    <premis:linkingAgentIdentifier>
+      <premis:linkingAgentIdentifierType valueURI="http://id.loc.gov/vocabulary/identifiers/local">url</premis:linkingAgentIdentifierType>
+      <premis:linkingAgentIdentifierValue>https://github.com/artefactual-sdps/preprocessing-sfa</premis:linkingAgentIdentifierValue>
+    </premis:linkingAgentIdentifier>
+  </premis:event>
+</premis:premis>
+`
+
+func TestAddPREMISEvent(t *testing.T) {
+	t.Parallel()
+
+	// Normal execution with no failures (for execution expected to work).
```

Comment:
Again, I think it's easier to read the tests if these test dirs are defined in the test structs.